pkg/proxy/backend/authenticator.go

```diff
@@ -152,6 +155,7 @@ func (auth *Authenticator) handshakeFirstTime(logger *zap.Logger, cctx ConnConte
 	auth.dbname = clientResp.DB
 	auth.collation = clientResp.Collation
 	auth.attrs = clientResp.Attrs
+	auth.capability &= clientResp.Capability
```

Comment:
Should be able to solve this at

https://github.com/pingcap/TiProxy/blob/1acb319b7fb56609d7268d7aacb4be753b2c92b6/pkg/proxy/backend/authenticator.go#L298-L317

Reply:
BTW: it seems that there is a bug that L299 and L300 should be swapped.